app/src/modules/location/location.c

```diff
@@ -54,14 +78,79 @@ void trigger_location_update(void)
 	}
 }
 
+void handle_network_chan(void) {
+	int err = 0;
+	enum network_status status;
+
+	if (gnss_initialized) {
+		return;
+	}
+
+	err = zbus_chan_read(&NETWORK_CHAN, &status, K_SECONDS(1));
+	if (err) {
+		LOG_ERR("Failed to read network status: %d", err);
+		SEND_FATAL_ERROR();
+		return;
+	}
+
+	if (status == NETWORK_CONNECTED) {
+		/* GNSS has to be enabled after the modem is initialized and enabled */
```

Comment:
For OOB this is fine, but for trackers with buffering capabilities we probably want to use GNSS before we have connectivity (i.e if we are out of coverage)

Reply:
Sure. This was just the easiest way to enable gnss when the modem was ready. The modem_init (or similarly named) callback was triggered before the modem was ready to enable gnss.